### DISCUSSION TITLE
How can I style local links separately from external links?

### DISCUSSION THREAD

**Discussion by maxigaz** [typst/typst#3170] [Q&A] [answered]
I would like to create the following rule:

- If a link is pointing to an external URL (such as a webpage), the link's body should be blue and underlined.
- If a link is pointing to a label, the body should be blue.

So far I've tried writing a show rule with a regex, but all links are underlined, so something is wrong.

```typ
#show link: it => {
  set text(blue)
  if it.dest == regex("^<*.>$") {
    it
  }
  else {
    underline(it)
  }
}

#link("https://github.com/")[Link to a webpage]

#link(<label>)[Link to label]

Some label here <label>
```

**Comment by MDLC01** [comment, marked as answer]
This should work:
```typ
#show link: it => {
  set text(blue)
  if type(it.dest) != str {
    it
  }
  else {
    underline(it)
  }
}
```
